tests/admin_scripts/tests.py

```diff
@@ -2160,6 +2160,20 @@ def test_overlaying_app(self):
             "won't replace conflicting files."
         )
 
+    def test_app_path(self):
+        args = ['startapp', 'test_app', 'app_parent/test_app']
+        app_parent = os.path.join(self.test_dir, 'app_parent')
+        test_app = os.path.join(app_parent, 'test_app')
+
+        os.mkdir(app_parent)
+        os.mkdir(test_app)
+        self.run_django_admin(args)
+        with open(os.path.join(test_app, 'apps.py')) as f:
+            content = f.read()
+            self.assertIn("class TestAppConfig(AppConfig)", content)
+            self.assertIn("name = 'app_parent.test_app'", content)
+        shutil.rmtree(app_parent)
```

Comment:
We should add test (with `self.subTest()`) for other scenarios, e.g.
- `app_parent/test_other` -> `app_parent.test_other`,
- `../test_project/app_parent/test_app` -> `app_parent.test_app`,
- `../test_other` should fallback to `test_app`.